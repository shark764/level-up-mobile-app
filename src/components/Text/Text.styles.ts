import tailwind from '@utils/tailwind';

const styles = {
  'display-1': tailwind('font-martian-extralight text-display-1'),
  'display-2': tailwind('font-martian-extralight text-display-2'),
  'heading-1': tailwind('font-martian-bold text-h1'),
  'heading-2': tailwind('font-martian-bold text-h2'),
  'heading-3': tailwind('font-martian-bold text-h3'),
  'heading-4': tailwind('font-martian-bold text-h4'),
  'heading-5': tailwind('font-martian-semibold text-h5'),
  'heading-6': tailwind('font-martian-medium text-h6 uppercase'),
  paragraph: tailwind('font-martian-extralight text-p'),
  body: tailwind('font-martian text-body'),
  'body-sm': tailwind('font-martian font-martian text-body-sm'),
  'body-md': tailwind('font-martian text-body-md'),
  'body-lg': tailwind('font-martian text-body-lg'),
  'body-semi': tailwind('font-martian-semibold text-body'),
  'body-sm-semi': tailwind('font-martian-semibold text-body-sm'),
  'body-md-semi': tailwind('font-martian-semibold text-body-md'),
  'body-lg-semi': tailwind('font-martian-semibold text-body-lg')
};

export default styles;
