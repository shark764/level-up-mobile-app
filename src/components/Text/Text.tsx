import React, { ReactNode } from 'react';
import { StyleProp, Text as T, TextStyle } from 'react-native';
import myStyles from './Text.styles';

interface Props {
  style?: StyleProp<TextStyle>;
  children: ReactNode;
  type:
    | 'display-1'
    | 'display-2'
    | 'heading-1'
    | 'heading-2'
    | 'heading-3'
    | 'heading-4'
    | 'heading-5'
    | 'heading-6'
    | 'paragraph'
    | 'body'
    | 'body-semi'
    | 'body-sm'
    | 'body-sm-semi'
    | 'body-md'
    | 'body-md-semi'
    | 'body-lg'
    | 'body-lg-semi';
}

const Text = ({ type, style = {}, children }: Props) => {
  return <T style={[myStyles[type], style]}>{children}</T>;
};

export default Text;
