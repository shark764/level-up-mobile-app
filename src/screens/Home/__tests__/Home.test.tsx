import React from 'react';
import { Home } from '../index';
import renderer from 'react-test-renderer';
import { createStackNavigator } from '@react-navigation/stack';
import { NavigationContainer } from '@react-navigation/native';
import { Text } from '@components/Text';
const { Navigator, Screen } = createStackNavigator();

const MockLogin = () => <Text type='body-lg-semi'>Login Screen</Text>;
const MockSignUp = () => <Text type='body-lg-semi'>SignUp Screen</Text>;

describe('<Home />', () => {
  it('renders correctly', async () => {
    const result = renderer
      .create(
        <NavigationContainer>
          <Navigator headerMode='none'>
            <Screen name='Home' component={Home} />
            <Screen name='Login' component={MockLogin} />
            <Screen name='SignUp' component={MockSignUp} />
          </Navigator>
        </NavigationContainer>
      )
      .toJSON();
    expect(result).toMatchSnapshot();
  });
  /*it('should show Login component when Log In button is pressed', () => {});
  it('should show SignUp component when Sign Up button is pressed', () => {});*/
});
