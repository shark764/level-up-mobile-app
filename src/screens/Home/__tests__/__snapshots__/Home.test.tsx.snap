// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      accessibilityIgnoresInvertColors={true}
                      style={
                        Object {
                          "flex": 1,
                          "height": "100%",
                          "justifyContent": "center",
                          "resizeMode": "cover",
                        }
                      }
                    >
                      <Image
                        source={
                          Object {
                            "testUri": "../../../src/assets/images/home-background.png",
                          }
                        }
                        style={
                          Array [
                            Object {
                              "bottom": 0,
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            },
                            Object {
                              "height": "100%",
                              "width": undefined,
                            },
                            undefined,
                          ]
                        }
                      />
                      <RCTSafeAreaView
                        emulateUnlessSupported={true}
                        style={
                          Array [
                            Object {
                              "height": "100%",
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                            undefined,
                          ]
                        }
                      >
                        <View
                          style={
                            Object {
                              "paddingLeft": 16,
                              "paddingRight": 16,
                            }
                          }
                        >
                          <RCTScrollView>
                            <View>
                              <View
                                style={
                                  Object {
                                    "paddingTop": 160,
                                  }
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "marginBottom": 112,
                                      },
                                    ]
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "marginBottom": 20,
                                      }
                                    }
                                  >
                                    <Text
                                      style={
                                        Array [
                                          Object {
                                            "fontFamily": "MartianBThai-Bold",
                                            "fontSize": 36,
                                            "lineHeight": 46.8,
                                          },
                                          Array [
                                            Object {
                                              "color": "rgba(255, 255, 255, 1)",
                                              "fontSize": 36,
                                              "lineHeight": 40,
                                            },
                                          ],
                                        ]
                                      }
                                    >
                                      Welcome
                                    </Text>
                                    <Text
                                      style={
                                        Array [
                                          Object {
                                            "fontFamily": "MartianBThai-Bold",
                                            "fontSize": 36,
                                            "lineHeight": 46.8,
                                          },
                                          Array [
                                            Object {
                                              "color": "rgba(255, 255, 255, 1)",
                                              "fontSize": 36,
                                              "lineHeight": 40,
                                            },
                                          ],
                                        ]
                                      }
                                    >
                                      Level Up Live
                                    </Text>
                                  </View>
                                  <Text
                                    style={
                                      Array [
                                        Object {
                                          "fontFamily": "MartianBThai-Bold",
                                          "fontSize": 20,
                                          "lineHeight": 31.2,
                                        },
                                        Object {
                                          "color": "rgba(198, 214, 238, 1)",
                                          "fontFamily": "MartianBThai-Regular",
                                        },
                                      ]
                                    }
                                  >
                                    Level up your game and
                                  </Text>
                                  <Text
                                    style={
                                      Array [
                                        Object {
                                          "fontFamily": "MartianBThai-Bold",
                                          "fontSize": 20,
                                          "lineHeight": 31.2,
                                        },
                                        Object {
                                          "color": "rgba(198, 214, 238, 1)",
                                          "fontFamily": "MartianBThai-Regular",
                                        },
                                      ]
                                    }
                                  >
                                    be part of the next-gen
                                  </Text>
                                  <Text
                                    style={
                                      Array [
                                        Object {
                                          "fontFamily": "MartianBThai-Bold",
                                          "fontSize": 20,
                                          "lineHeight": 31.2,
                                        },
                                        Object {
                                          "color": "rgba(198, 214, 238, 1)",
                                          "fontFamily": "MartianBThai-Regular",
                                        },
                                      ]
                                    }
                                  >
                                    shooters
                                  </Text>
                                </View>
                                <View
                                  style={
                                    Object {
                                      "backgroundColor": "#6200ee",
                                      "borderBottomLeftRadius": 8,
                                      "borderBottomRightRadius": 8,
                                      "borderColor": "transparent",
                                      "borderRadius": 4,
                                      "borderStyle": "solid",
                                      "borderTopLeftRadius": 8,
                                      "borderTopRightRadius": 8,
                                      "borderWidth": 0,
                                      "elevation": 2,
                                      "marginBottom": 8,
                                      "minWidth": 64,
                                      "shadowColor": "#000000",
                                      "shadowOffset": Object {
                                        "height": 0.75,
                                        "width": 0,
                                      },
                                      "shadowOpacity": 0.24,
                                      "shadowRadius": 1.5,
                                    }
                                  }
                                  title="Sign Up"
                                >
                                  <View
                                    accessibilityRole="button"
                                    accessibilityState={
                                      Object {
                                        "disabled": false,
                                      }
                                    }
                                    accessible={true}
                                    focusable={true}
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      Array [
                                        Object {
                                          "overflow": "hidden",
                                        },
                                        Object {
                                          "borderRadius": 4,
                                        },
                                      ]
                                    }
                                  >
                                    <View
                                      style={
                                        Array [
                                          Object {
                                            "alignItems": "center",
                                            "flexDirection": "row",
                                            "justifyContent": "center",
                                          },
                                          Array [
                                            Object {
                                              "borderBottomLeftRadius": 8,
                                              "borderBottomRightRadius": 8,
                                              "borderTopLeftRadius": 8,
                                              "borderTopRightRadius": 8,
                                            },
                                            Object {
                                              "backgroundColor": "rgba(100, 97, 236, 1)",
                                            },
                                            Object {},
                                            undefined,
                                          ],
                                        ]
                                      }
                                    >
                                      <Text
                                        numberOfLines={1}
                                        selectable={false}
                                        style={
                                          Array [
                                            Object {
                                              "color": "#000000",
                                              "fontFamily": "System",
                                              "fontWeight": "400",
                                            },
                                            Object {
                                              "textAlign": "left",
                                            },
                                            Array [
                                              Object {
                                                "letterSpacing": 1,
                                                "marginHorizontal": 16,
                                                "marginVertical": 9,
                                                "textAlign": "center",
                                              },
                                              undefined,
                                              Object {
                                                "textTransform": "uppercase",
                                              },
                                              Object {
                                                "color": "#ffffff",
                                                "fontFamily": "System",
                                                "fontWeight": "500",
                                              },
                                              Object {
                                                "fontFamily": "System",
                                                "fontWeight": "500",
                                              },
                                              undefined,
                                            ],
                                          ]
                                        }
                                      >
                                        <Text
                                          style={
                                            Array [
                                              Object {
                                                "fontFamily": "MartianBThai-Regular",
                                                "fontSize": 14,
                                                "lineHeight": 23.8,
                                              },
                                              Array [
                                                Object {
                                                  "textTransform": "none",
                                                },
                                                Object {
                                                  "color": "rgba(255, 255, 255, 1)",
                                                },
                                                undefined,
                                              ],
                                            ]
                                          }
                                        >
                                          Sign Up
                                        </Text>
                                      </Text>
                                    </View>
                                  </View>
                                </View>
                                <View
                                  style={
                                    Object {
                                      "backgroundColor": "transparent",
                                      "borderBottomLeftRadius": 8,
                                      "borderBottomRightRadius": 8,
                                      "borderColor": "transparent",
                                      "borderRadius": 4,
                                      "borderStyle": "solid",
                                      "borderTopLeftRadius": 8,
                                      "borderTopRightRadius": 8,
                                      "borderWidth": 0,
                                      "elevation": 0,
                                      "marginBottom": 8,
                                      "minWidth": 64,
                                    }
                                  }
                                  title="Log In"
                                  titleStyle={
                                    Object {
                                      "color": "rgba(255, 255, 255, 1)",
                                    }
                                  }
                                >
                                  <View
                                    accessibilityRole="button"
                                    accessibilityState={
                                      Object {
                                        "disabled": false,
                                      }
                                    }
                                    accessible={true}
                                    focusable={true}
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      Array [
                                        Object {
                                          "overflow": "hidden",
                                        },
                                        Object {
                                          "borderRadius": 4,
                                        },
                                      ]
                                    }
                                  >
                                    <View
                                      style={
                                        Array [
                                          Object {
                                            "alignItems": "center",
                                            "flexDirection": "row",
                                            "justifyContent": "center",
                                          },
                                          Array [
                                            Object {
                                              "borderBottomLeftRadius": 8,
                                              "borderBottomRightRadius": 8,
                                              "borderTopLeftRadius": 8,
                                              "borderTopRightRadius": 8,
                                            },
                                            Object {
                                              "backgroundColor": "transparent",
                                            },
                                            Object {},
                                            undefined,
                                          ],
                                        ]
                                      }
                                    >
                                      <Text
                                        numberOfLines={1}
                                        selectable={false}
                                        style={
                                          Array [
                                            Object {
                                              "color": "#000000",
                                              "fontFamily": "System",
                                              "fontWeight": "400",
                                            },
                                            Object {
                                              "textAlign": "left",
                                            },
                                            Array [
                                              Object {
                                                "letterSpacing": 1,
                                                "marginHorizontal": 16,
                                                "marginVertical": 9,
                                                "textAlign": "center",
                                              },
                                              undefined,
                                              Object {
                                                "textTransform": "uppercase",
                                              },
                                              Object {
                                                "color": "#6200ee",
                                                "fontFamily": "System",
                                                "fontWeight": "500",
                                              },
                                              Object {
                                                "fontFamily": "System",
                                                "fontWeight": "500",
                                              },
                                              undefined,
                                            ],
                                          ]
                                        }
                                      >
                                        <Text
                                          style={
                                            Array [
                                              Object {
                                                "fontFamily": "MartianBThai-Regular",
                                                "fontSize": 14,
                                                "lineHeight": 23.8,
                                              },
                                              Array [
                                                Object {
                                                  "textTransform": "none",
                                                },
                                                Object {
                                                  "color": "rgba(20, 22, 43, 1)",
                                                },
                                                Object {
                                                  "color": "rgba(255, 255, 255, 1)",
                                                },
                                              ],
                                            ]
                                          }
                                        >
                                          Log In
                                        </Text>
                                      </Text>
                                    </View>
                                  </View>
                                </View>
                                <View
                                  style={
                                    Object {
                                      "alignItems": "center",
                                      "marginTop": 80,
                                    }
                                  }
                                >
                                  <Text
                                    style={
                                      Array [
                                        Object {
                                          "fontFamily": "MartianBThai-Regular",
                                          "fontSize": 14,
                                          "lineHeight": 23.8,
                                        },
                                        Object {
                                          "color": "rgba(144, 241, 219, 1)",
                                        },
                                      ]
                                    }
                                  >
                                    By using Level Up Live you accept
                                  </Text>
                                  <Text
                                    style={
                                      Array [
                                        Object {
                                          "fontFamily": "MartianBThai-Regular",
                                          "fontSize": 14,
                                          "lineHeight": 23.8,
                                        },
                                        Object {
                                          "color": "rgba(144, 241, 219, 1)",
                                        },
                                      ]
                                    }
                                  >
                                    the Terms of Service and Privacy Policy
                                  </Text>
                                </View>
                              </View>
                            </View>
                          </RCTScrollView>
                        </View>
                      </RCTSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
